File: php/src/Domain/Purchase/MainRepository.php
<?php

namespace Katalogs\BookInvoicing\Domain\Purchase;

class MainRepository
{
    private static Repository $runningRepository;

    public static function configuredRepository(): Repository
    {
        if (self::$runningRepository == null) {
            self::$runningRepository = new JsonRepository();
        }
        return self::$runningRepository;
    }

    // TESTING ONLY
    public static function override(Repository $newRepository): void {
        self::$runningRepository = $newRepository;
    }

    // TESTING ONLY
    public static function reset(): void {
        self::$runningRepository = new JsonRepository();
    }
}

File: java/src/main/java/org/peaksys/gameoftechs/MainRepository.java
package org.peaksys.gameoftechs;

import org.peaksys.gameoftechs.storage.JsonRepository;
import org.peaksys.gameoftechs.storage.Repository;

public final class MainRepository {

    private static Repository runningRepository;

    private MainRepository() {
    }

    public static Repository configuredRepository() {
        if (runningRepository == null) {
            runningRepository = new JsonRepository();
        }
        return runningRepository;
    }

    /* Working effectively with Legacy Code
     * https://www.goodreads.com/book/show/44919.Working_Effectively_with_Legacy_Code
     */
    // TESTING ONLY
    public static void override(Repository newRepository) {
        runningRepository = newRepository;
    }

    // TESTING ONLY
    public static void reset() {
        runningRepository = new JsonRepository();
    }
}
